src/outagesim/backup_reliability.jl

```diff
@@ -0,0 +1,925 @@
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+# IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+# INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+# LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
+# OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+# OF THE POSSIBILITY OF SUCH DAMAGE.
+# *********************************************************************************
+Numeric = Union{Int, Float64}
+
+"""
+    transition_prob(n::Vector{Int}, n_prime::Vector{Int}, p::Numeric)
+
+Return the probability of going from i to j generators given a failure rate of ``p`` for each i,j pair in vectors ``n`` and ``n_prime``.
+
+Function used to create transition probabilities in Markov matrix.
+
+# Examples
+```repl-julia
+julia> transition_prob([1,2,3,4], [0,1,2,3], 0.5)
+4-element Vector{Float64}:
+ 0.5
+ 0.5
+ 0.375
+ 0.25
+```
+"""
+function transition_prob(n::Vector{Int}, n_prime::Vector{Int}, p::Numeric)::Vector{Float64} 
+    return binomial.(n, n_prime).*(1-p).^(n_prime).*(p).^(n-n_prime)
+end
+
+
+"""
+    markov_matrix(N::Int, p::Numeric)
+
+Return an ``N``+1 by ``N``+1 matrix of transition probabilities of going from n (row) to n' (column) given probability ``p``
+
+Row n denotes starting with n-1 generators, with the first row denoting zero working generators. Column n' denots ending with n'-1 generators.
+
+# Examples
+```repl-julia
+julia> markov_matrix(2, 0.1)
+3×3 Matrix{Float64}:
+ 1.0   0.0   0.0
+ 0.1   0.9   0.0
+ 0.01  0.18  0.81
+```
+"""
+function markov_matrix(N::Int, p::Numeric)::Matrix{Float64} 
+    #Creates Markov matrix for generator transition probabilities
+    M = reshape(transition_prob(repeat(0:N, outer = N + 1), repeat(0:N, inner = N+1), p), N+1, N+1)
+    replace!(M, NaN => 0)
+    return M
+end
+
+"""
+    starting_probabilities(N::Int, OA::Numberic, FTS::Numeric)
+
+Return a 1 by ``N``+1 by matrix (row vector) of probabilities of number of generators operationally available (``OA``) and avoiding
+a Failure to Start (``FTS``)
+
+The first element denotes no generators successfully starts and element n denotes n-1 generators start
+
+# Arguments
+- `N::Int`: the number of generators 
+- `OA::Numeric`: Operational Availability. The chance that a generator will be available (not down for maintenance) at the start of the outage
+- `FTS::Numeric`: Failure to Start. The chance that a generator fails to successfully start and take load.
+
+# Examples
+```repl-julia
+julia> starting_probabilities(2, 0.99, 0.05)
+1×3 Matrix{Float64}:
+ 0.00354025  0.11192  0.88454
+```
+"""
+function starting_probabilities(N::Int, OA::Numeric, FTS::Numeric)::Matrix{Float64} 
```

Comment:
please use descriptive names for variables. it is nice to keep them consistent from user input names (is `FTS` here the same as `gen_fts` provided by the user?)